central/deployment/store/cache/cache.go

```diff
@@ -6,201 +6,161 @@ import (
 	"github.com/stackrox/rox/central/deployment/store"
 	"github.com/stackrox/rox/central/deployment/store/types"
 	"github.com/stackrox/rox/generated/storage"
-	"github.com/stackrox/rox/pkg/logging"
-	"github.com/stackrox/rox/pkg/size"
-	"github.com/stackrox/rox/pkg/sizeboundedcache"
-	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
-const (
-	maxCachedDeploymentSize = 50 * size.KB  // if it's larger than 50KB then we aren't going to cache it
-	maxCacheSize            = 200 * size.MB // 200 MB
-)
+// NewCachedStore caches the deployment store
+func NewCachedStore(store store.Store) (store.Store, error) {
+	impl := &cacheImpl{
+		store: store,
 
-var (
-	log = logging.LoggerForModule()
-)
+		cache: make(map[string]*storage.Deployment),
+	}
 
-type deploymentTombstone struct{}
+	if err := impl.populate(); err != nil {
+		return nil, err
+	}
+	return impl, nil
+}
 
-func sizeFunc(k, v interface{}) int64 {
-	if dep, ok := v.(*storage.Deployment); ok {
-		return int64(len(k.(string)) + dep.Size())
+func (c *cacheImpl) GetListDeployment(ctx context.Context, id string) (*storage.ListDeployment, bool, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	deployment, ok := c.cache[id]
+	if !ok {
+		return nil, false, nil
 	}
-	return int64(len(k.(string)))
+	return types.ConvertDeploymentToDeploymentList(deployment), true, nil
 }
 
-// NewCachedStore returns an deployment store implementation that caches deployments
-func NewCachedStore(store store.Store) store.Store {
-	// This is a size based cache, where we use LRU to determine which of the oldest elements should
-	// be removed to allow a new element
-	cache, err := sizeboundedcache.New(maxCacheSize, maxCachedDeploymentSize, sizeFunc)
-	utils.CrashOnError(err)
+func (c *cacheImpl) GetManyListDeployments(ctx context.Context, ids ...string) ([]*storage.ListDeployment, []int, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
 
-	return &cachedStore{
-		store: store,
-		cache: cache,
+	listDeployments := make([]*storage.ListDeployment, 0, len(ids))
+	var missingIndices []int
+	for i, id := range ids {
+		deployment, ok := c.cache[id]
+		if !ok {
+			missingIndices = append(missingIndices, i)
+			continue
+		}
+		listDeployments = append(listDeployments, types.ConvertDeploymentToDeploymentList(deployment))
 	}
+	return listDeployments, missingIndices, nil
 }
 
-// This cached store implementation relies on the usage of the Deployment store so this may not be easily portable
-// to other sections of the code.
-type cachedStore struct {
+type cacheImpl struct {
 	store store.Store
-	cache sizeboundedcache.Cache
+
+	cache map[string]*storage.Deployment
+	lock  sync.RWMutex
+}
+
+func (c *cacheImpl) addNoLock(deployment *storage.Deployment) {
+	c.cache[deployment.GetId()] = deployment.Clone()
```

Comment:
probably nbd but you could rewrite the code to have the `Clone()` be performed outside of the critical section